zng/docs/zson.md

```diff
@@ -506,52 +536,76 @@ map type, or two elements referring to the type of the keys and type of the valu
 A map value may be empty.  An empty map value without a type decorator is
 presumed to be an empty map of type (`null`, `null`).
 
-### 3.5 Type Value
+#### 3.4.7 Type Value
 
-The syntax of a type value or of a decorator expressed as a type value mirrors
-the value syntax.
+The type of a type value is `type` while its value is depicted by parenthesizing
+its type description (as defined in [Section 3.5](#35-type-syntax).  For example,
```

Comment:
There's a yaml test elsewhere in this PR that has: 
```
  {
      typeof: ({x:int32,s:string}),
      count: 2 (uint64)
  } (=0)
```

When I first read this, trying to find a pattern and knowing that the parenthesized `(uint64)` , my first guess was that `({x:int32,s:string})` was also a type...  but that couldn't be since there was no value.

So I'm wondering what is the motivation for parentheizing type values... is there some syntactical ambiguity?

Anyway, more of a design quetion than code review... so don't let it hold up the PR.

Reply:
Yes, the zson parser is simpler with type values beginning with open paren as no value begins with that character.  Otherwise, you need deep backtracking to differentiate between 
```
{ foo:int64 }
```
as a type value and 
```
{ foo:10 }
```
as a value.  I also thought using the decorator parens for type value parens unified the look and feel but I see that might be a bit confusing. 

With the open paren here, I think the parser needs just one rune of lookahead throughout but I just started working on the parser and we will see.  If something unexpected comes up, maybe we can revisit this?